lib/final_destination.rb

```diff
@@ -397,7 +398,11 @@ def validate_uri
   def validate_uri_format
     return false unless @uri && @uri.host
     return false unless %w[https http].include?(@uri.scheme)
-    return false if @uri.scheme == "http" && @uri.port != 80
+
+    # In development or test environments we may want to point to a http URL which is local
+    # on a different port than 80 (e.g. for local S3 providers like minio)
+    return false if @uri.scheme == "http" && (Rails.env.production? && @uri.port != 80)
```

Comment:
Production-specific logic isn't ideal. How about adding a very specific bypass when the hostname is included in `SiteSetting.allowed_internal_hosts`, and it matches the host/port of `SiteSetting.s3_endpoint` ?

Reply:
Ok will give that a shot, it does sound nicer and I didn't like the production thing either, ty :)